Adapter/Sources/audioplayer.cpp

```diff
@@ -0,0 +1,43 @@
+#include <iostream>
+
+#include "audioplayer.h"
+
+MediaAdapter::MediaAdapter(const std::string& audioFormat)
+{
+	if (audioFormat == "mp4")
+	{
+		m_advanceMediaPlayer = new Mp4Player();
+	}
+	else if (audioFormat == "vlc") {
+		m_advanceMediaPlayer = new VlcPlayer();
+	}
+}
+
+void MediaAdapter::play(const std::string& audioFormat, const std::string& fileName)
+{
+	if (audioFormat == "mp4")
+	{
+		m_advanceMediaPlayer->playMp4(fileName);
+	}
+	else if (audioFormat == "vlc") {
+		m_advanceMediaPlayer->playVlc(fileName);
+	}
+}
+
+void AudioPlayer::play(const std::string& audioFormat, const std::string& fileName)
+{
+	if (audioFormat == "mp3")
+	{
+		std::cout << "Playing mp3 file. Name : " << fileName << std::endl;
+		return;
+	}
+	if (audioFormat == "mp4" || audioFormat == "vlc")
+	{
+		m_mediaAdapter = new MediaAdapter(audioFormat);
```

Comment:
do we need to pass `audioFormat` twice to `m_mediaAdapter` object?
I mean is it expected to call `playMp4` method on `VlcPlayer` object?

if no it would be better to keep `audioFormat` as a member in `mediaAdapter` and use it in `play` method instead of passing it here (`m_mediaAdapter->play(**audioFormat**, fileName);`)